src/tools/compiletest/src/runtest.rs

```diff
@@ -3784,6 +3784,7 @@ impl<'test> TestCx<'test> {
             .env("TMPDIR", &tmpdir)
             .env("LD_LIB_PATH_ENVVAR", dylib_env_var())
             .env("HOST_RPATH_DIR", cwd.join(&self.config.compile_lib_path))
+            .env("LD_LIBRARY_PATH", cwd.join(&self.config.compile_lib_path))
```

Comment:
Hmm, I wonder if this is the right place to solve it. The original run-make specifies `LD_LIBRARY_PATH` via `LD_LIB_PATH_ENVVAR` (when that is the correct `LD_LIB_PATH_ENVVAR` for the target):

https://github.com/rust-lang/rust/blob/7f3a62b17b3d72746c8c839baca851bac7544180/src/tools/compiletest/src/runtest.rs#L3582

In `tools.mk`, then `LD_LIBRARY_PATH` (which is the suitable `LD_LIB_PATH_ENVVAR`) is set to the correct paths

https://github.com/rust-lang/rust/blob/536606bc5df7e6045c6dbedc2f852ba3d38581f1/tests/run-make/tools.mk#L3-L6

My guess is that

https://github.com/rust-lang/rust/blob/536606bc5df7e6045c6dbedc2f852ba3d38581f1/src/tools/run-make-support/src/lib.rs#L142-L154

is broken, but this should be the correct place to fix it. In particular, I think you'd just need to preserve the semantics of

```make
 HOST_RPATH_ENV = \ 
     $(LD_LIB_PATH_ENVVAR)="$(TMPDIR):$(HOST_RPATH_DIR):$($(LD_LIB_PATH_ENVVAR))" 
 TARGET_RPATH_ENV = \ 
     $(LD_LIB_PATH_ENVVAR)="$(TMPDIR):$(TARGET_RPATH_DIR):$($(LD_LIB_PATH_ENVVAR))"
```

Reply:
Ah, I think `"TARGET_RPATH_ENV"` should instead be `TARGET_RPATH_DIR`? Can you check if that fixes the issue?

Reply:
@jieyouxu Updating TARGET_RPATH_ENV with TARGET_RAPTH_DIR doesn't work on sgx.

Reply:
Hmm nevermind then. I see the original run-make test also had this issue...

Reply:
Actually, could you try changing

```rs
- for p in env::split_paths(&env::var("TARGET_RPATH_ENV").unwrap()) { 
-     paths.push(p.to_path_buf()); 
- }
+ for p in env::split_paths(&env::var("HOST_RPATH_DIR").unwrap()) { 
+     paths.push(p.to_path_buf()); 
+ }
+ for p in env::split_paths(&env::var("TARGET_RPATH_DIR").unwrap()) { 
+     paths.push(p.to_path_buf()); 
+ }
```

Judging from https://github.com/rust-lang/rust/pull/113889, `HOST_RPATH_DIR` should have the paths you need (I hope)

Reply:
Fixing this with something like this makes sense to me.